pinot-core/src/main/java/org/apache/pinot/core/common/ObjectSerDeUtils.java

```diff
@@ -123,7 +124,9 @@ public enum ObjectType {
     FloatLongPair(29),
     DoubleLongPair(30),
     StringLongPair(31),
-    CovarianceTuple(32);
+    CovarianceTuple(32),
```

Comment:
i was wondering if we can simply reuse CovarianceTuple and logic here. 
basically Variance(x) = Covarance(x, x)

Reply:
If I reuse the covariance implementation, then it will become `naive algorithm` on https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Na%C3%AFve_algorithm

According to the documentation, directly computing `Var = (SumSq − (Sum × Sum) / n) / (n − 1)` is not numerically stable. 

The existing implementation is being used by Hive and Presto and they look to have separate implementation for covariance and variance.

Reply:
sweet! thanks for the clarification. this also means there's numeric stability for covariance. probably we should also document that. (or change the cover algo).

Reply:
we can indeed change the covar algo to numerical stable version: similar to 
 https://github.com/trinodb/trino/blob/1866a23e3b0377144c1820de892c0de2762351a8/core/trino-main/src/main/java/io/trino/operator/aggregation/state/CovarianceState.java